FirebaseSessions/Sources/Public/SessionsDependencies.swift

```diff
@@ -15,18 +15,62 @@
 
 import Foundation
 
-// Sessions Dependencies determines when a dependent SDK is
-// installed in the app. The Sessions SDK uses this to figure
-// out which dependencies to wait for to getting the data
-// collection state.
-//
-// This is important because the Sessions SDK starts up before
-// dependent SDKs
+private final class AtomicBox<T> {
+  private var _value: T
+  private let lock = NSLock()
+
+  init(_ value: T) {
+    _value = value
+  }
+
+  func value() -> T {
+    lock.withLock {
+      _value
+    }
+  }
+
+  @discardableResult func withLock(_ body: (_ value: inout T) -> Void) -> T {
+    lock.withLock {
+      body(&_value)
+      return _value
+    }
+  }
+}
+
+/// Sessions Dependencies determines when a dependent SDK is
+/// installed in the app. The Sessions SDK uses this to figure
+/// out which dependencies to wait for to getting the data
+/// collection state.
+///
+/// This is important because the Sessions SDK starts up before
+/// dependent SDKs
 @objc(FIRSessionsDependencies)
 public class SessionsDependencies: NSObject {
-  static var dependencies: Set<SessionsSubscriberName> = .init()
+  #if compiler(>=6)
+    private nonisolated(unsafe) static let _dependencies: AtomicBox<Set<SessionsSubscriberName>> =
+      AtomicBox(
+        Set()
+      )
+  #else
+    private static let _dependencies: AtomicBox<Set<SessionsSubscriberName>> = AtomicBox(
+      Set()
+    )
+  #endif
+
+  static var dependencies: Set<SessionsSubscriberName> {
+    _dependencies.value()
+  }
 
   @objc public static func addDependency(name: SessionsSubscriberName) {
-    SessionsDependencies.dependencies.insert(name)
+    _dependencies.withLock { dependencies in
```

Comment:
Out of curious how Core initialize the SDKs? I think currently this function is only used by Crashlytics and Perf when each SDK trying to register themselves to Session. Are those two SDKs can be potentially initialized in parallel or this change is just for the purpose of reduce warnings?

Reply:
> Out of curious how Core initialize the SDKs? 

Core initializes them as part of the component system– https://github.com/firebase/firebase-ios-sdk/blob/main/Interop/FirebaseComponentSystem.md#registering-with-core

> Are those two SDKs can be potentially initialized in parallel or this change is just for the purpose of reduce warnings?

This change is just to reduce the warnings. There may be a practical race condition this avoids depending on how  Crashlytics and Perf use this API.

Reply:
Yeah I have realized the component register system but just want to double check the behaviour since the doc does not indicate the if `componentsToRegister:` is call serially for each SDK. Sounds all good then.